### ISSUE TITLE
Zarf package file is set to 0 bytes on disconnect from install\update session

### ISSUE THREAD

**Issue by tmahalligan** [defenseunicorns/zarf#2339]
### Environment
Device and OS: EC2\Amazon Linux 2 
App version: 0.31.2
Kubernetes distro being used: EKS
Other:

### Steps to reproduce
Run Zarf deployment and have session disconnected before completion 

### Expected result
After disconnection from deploy\update the package file is in place with proper size.

### Actual Result
After disconnection from deploy\update the package file is 0 bytes

### Visual Proof (screenshots, videos, text, etc)
![Screenshot 2024-02-23 at 2 42 42 PM](https://github.com/defenseunicorns/zarf/assets/7749768/4f6bc78f-8017-48c9-93a8-6da894f0ced5)

### Severity/Priority
Low (We have work around)

### Additional Context
I noted when doing an initial zarf deploy, or a zarf version update,  the total time of the deployment will sometimes exceed whatever SSM console or SSH timeout value that we might be working with

When the deployment is interrupted by one of these timeouts the concatenated zarf package deployment file which should be ~12GB will have a zero byte value and the engineer will need to pull the part files to their local filesystem to rerun deployment, which then invariably succeeds.

We work around this issue by running deploys in screen or tmux but I was curious if this is know bug

**Comment by Noxsios** [comment]
@tmahalligan By `concatenated zarf package` are your referring to split packages via `--max-package-size`?

**Comment by tmahalligan** [comment]
Am sorry for late reply @Noxsios we have a multi-part package that is joined together on successful deploy leaving a single large package file which we the use for deployments to other clusters. We had noticed that on a timeout (SSM or SSH) during deploy when we reestablished as session there would be a single 0 byte file and we were forced to pull the parts again.

I have been thinking on it and I am wondering if this might be an ID10T error where upon reestablishing connectivity the user up arrows to the last command which references the part000 file that is no longer in place and that zeros out the big single file. I will test that and follow back up if that is the case or not.

Thanks for the follow-up
